ksops.go

```diff
@@ -47,30 +87,56 @@ func main() {
 		os.Exit(1)
 	}
 
-	if manifest.Files == nil {
-		fmt.Fprintf(os.Stderr, "missing the required 'files' key in the ksops manifests: %s", content)
+	if manifest.Files == nil && manifest.SecretFrom == nil {
+		fmt.Fprintf(os.Stderr, "missing the required 'files' or 'secretFrom' key in the ksops manifests: %s", content)
 		os.Exit(1)
 	}
 
 	var output bytes.Buffer
 
 	for _, file := range manifest.Files {
-		b, err := ioutil.ReadFile(file)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error reading %q: %q\n", file, err.Error())
-			fmt.Fprintf(os.Stderr, "manifest content: %s", content)
-			os.Exit(1)
+		data := DecryptFile(file, content)
+
+		output.Write(data)
+		output.WriteString("\n---\n")
+	}
+
+	for _, secretFrom := range manifest.SecretFrom {
+		stringData := make(map[string]string)
+
+		for _, file := range secretFrom.Files {
+			data := DecryptFile(file, content)
+
+			key := filepath.Base(file)
+			stringData[key] = string(data)
 		}
 
-		format := formats.FormatForPath(file)
-		data, err := decrypt.DataWithFormat(b, format)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "trouble decrypting file %s", err.Error())
-			os.Exit(1)
+		for _, file := range secretFrom.Envs {
+			data := DecryptFile(file, content)
+
+			env, err := godotenv.Unmarshal(string(data))
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error unmarshalling .env file %s", err.Error())
+				os.Exit(1)
+			}
+			for k, v := range env {
+				stringData[k] = v
+			}
 		}
 
-		output.Write(data)
-		output.WriteString("\n---\n")
+		s := KubernetesSecret{
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			Metadata:   secretFrom.Metadata,
+			Type:       secretFrom.Type,
+			StringData: stringData,
+		}
+		d, err := yaml.Marshal(&s)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "trouble encoding metadata %s", err.Error())
+		}
```

Comment:
Any reason not to abort on error here?

```suggestion
		if err != nil {
			fmt.Fprintf(os.Stderr, "trouble encoding metadata %s", err.Error())
			os.Exit(1)
		}
```

Reply:
No just the leftovers from the copypasta 😄